boofuzz/sessions.py

```diff
@@ -1505,13 +1512,26 @@ def _open_connection_keep_trying(self, target):
         """
         if not self._reuse_target_connection:
             out_of_available_sockets_count = 0
+            unable_to_connect_count = 0
+            initial_time = time.time()
+
             while True:
                 try:
                     target.open()
                     break  # break if no exception
                 except exception.BoofuzzTargetConnectionFailedError:
                     self._fuzz_data_logger.log_info(constants.WARN_CONN_FAILED_TERMINAL)
+                    if unable_to_connect_count == self.restart_threshold:
```

Comment:
If we change to `>=` (which I also prefer for safety), we'll also need another check to verify the value was actually set (with `==`, a default value of `-1` will never trigger the the threshold). The first idea that comes to mind is making the default None and doing:
```python
if self.restart_threshold and ...
```
This will keep the default approach consistent with None as a default, and make the check more explicit than it was with `==`.